modules/tensor_mechanics/src/materials/RadialReturnCreepStressUpdateBase.C

```diff
@@ -0,0 +1,69 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "RadialReturnCreepStressUpdateBase.h"
+
+template <>
+InputParameters
+validParams<RadialReturnCreepStressUpdateBase>()
+{
+  InputParameters params = validParams<RadialReturnStressUpdate>();
+
+  params.addCoupledVar("temperature", "Coupled temperature");
+  params.addDeprecatedParam<std::string>(
+      "creep_prepend",
+      "",
+      "String that is prepended to the creep_strain Material Property",
+      "This has been replaced by the 'base_name' parameter");
+  params.set<std::string>("effective_inelastic_strain_name") = "effective_creep_strain";
+
+  return params;
+}
+
+RadialReturnCreepStressUpdateBase::RadialReturnCreepStressUpdateBase(
+    const InputParameters & parameters)
+  : RadialReturnStressUpdate(parameters),
+    _creep_prepend(getParam<std::string>("creep_prepend")),
+    _has_temp(isCoupled("temperature")),
+    _temperature(_has_temp ? coupledValue("temperature") : _zero),
```

Comment:
It may have made sense for that particular creep model to default temperature to zero, but that's a bad idea in general for all creep models that will potentially derive from this class. Since temperature isn't even used at this level, I'd say move it up to the power law creep model and leave its default behavior as it was.

Reply:
Sure, it is a model parameter anyways. Keep the default at zero in the power law model?

Reply:
I would prefer to make it required, but I'm OK either way. I guess to minimize the impact of this commit on code behavior, it might be best to leave it alone.

Making it required isn't terribly dangerous to do because the worst thing that will happen is that someone's input file won't work, and then they'll go and add the missing parameter.  The nice thing is that you can supply a number (rather than a variable) for a parameter that accepts a coupled variable, so you can just set 'temperature=0' in the input file.

Reply:
Yeah, that behavior isn't even valid now since there is a `isCoupled` guard for temperature.